From f8859fbe310e57665aa50717edc1d4342d9a99d9 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 10 Feb 2016 17:55:03 -0800
Subject: [PATCH 669/827] drm: xilinx: fb: Check the ret value correctly

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Should wait for vblank when returned value from drm_crtc_vblank_get()
is 0.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 641faf479fef1d483c61680affb40465dcca7d9e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
index 8a2d7ce..9bd7c0f 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_fb.c
@@ -181,7 +181,7 @@ xilinx_drm_fb_ioctl(struct fb_info *info, unsigned int cmd, unsigned long arg)
 			mode_set = &fb_helper->crtc_info[i].mode_set;
 			crtc = mode_set->crtc;
 			ret = drm_crtc_vblank_get(crtc);
-			if (ret) {
+			if (!ret) {
 				drm_crtc_wait_one_vblank(crtc);
 				drm_crtc_vblank_put(crtc);
 			}
-- 
2.9.3

